pinot-common/src/main/java/org/apache/pinot/common/utils/FileUploadDownloadClient.java

```diff
@@ -747,6 +750,33 @@ public SimpleHttpResponse uploadSegment(URI uri, String segmentName, InputStream
     return uploadSegment(uri, segmentName, inputStream, null, parameters, HttpClient.DEFAULT_SOCKET_TIMEOUT_MS);
   }
 
+  public List<String> getSegments(URI uri, String rawTableName, String tableType, boolean excludeReplacedSegments)
+      throws URISyntaxException, IOException, HttpErrorStatusException {
+    ControllerRequestURLBuilder controllerRequestURLBuilder = ControllerRequestURLBuilder.baseUrl(uri.toString());
```

Comment:
Controller URIs are provided as input to PinotClusterSpec, which eventually make their way here, however, I'm not sure what's the proper way to convert URIs into URLs..